### ISSUE TITLE
Collaboration with CO: discussion with kanban

### ISSUE THREAD

**Issue by fosterlynn** [opencooperativeecosystem/kanban#12]
Briefly, this is a possibility to work with @oceatoon from CO (Communecter) to see if we can create interoperability between the kanban of @ivanminutillo and the DDA (discussion-decision-action) module that CO is using.  It could potentially provide discussion within the kanban, perhaps on a process or a commitment level.  On the CO side, it could potentially provide a kanban view of their actions.

From a telegram chat, with embedded images added here before the copied chat text:

CO:
![co-action](https://user-images.githubusercontent.com/3776081/35780481-b37b37f0-09a1-11e8-9680-39f8f53493f1.jpeg)
![co-pra](https://user-images.githubusercontent.com/3776081/35780482-b6140dac-09a1-11e8-9200-12990c4ead53.jpeg)

kanban:
![kanban-1](https://user-images.githubusercontent.com/3776081/35780486-c59bae1a-09a1-11e8-8a0b-b9a07004b118.png)
![kanban-2](https://user-images.githubusercontent.com/3776081/35780487-c80405f8-09a1-11e8-943a-6defb10cc7dc.png)

Lynn Foster
@oceatoon on the DDA module - can that be added on to existing webpages?  Like could you be looking at @Bernin1 's kanban page, click on an icon in a process, and have a discussion panel show up on the bottom of the page?  Or something like that?
or the organization stuff also strikes me as something we could look at plugging in to a pane on a page somehow?
(looking for a concrete experiment)
Bernini
Lynn Foster
(looking for a concrete experiment)
+1
Lynn Foster
or on the organization side, i wonder if we could experiment with the CO page and try to plug something like the kanban in there [if that makes sense, it would be in addition to the current kanban, which is more person-centered, covers all the organizations the person is part of]
but the DDA module would definitely speak to an immediate requirement, I like that one
Tib Kat
@fosterlynn sure, we can imagine whatever we like 
actually we talk about the kanban view on that module
it's great option
Lynn Foster
(looking for a concrete experiment)
great
I'd love to have access to the current state of the kanban
is you kanban homemade or based on seome existing kanban project ?
Bernini
made from scratch
Tib Kat
Lynn Foster
but the DDA module would definitely speak to an immediate requirement, I like that one
I agree as discussed , it's a collaboration space made to evolve 
it holds proposals and actions, which is close to part of Value Flows
cool :)
Bernini
made from scratch
cool
Bernini
I didnt like very much wekan, and it should be flexible enough to adabt both to a status view (todo - doing -done) than a resource flow :D
Tib Kat
it could indeed be something we could also use
which sounds like a great win win experiment ,no ?
and we have actions on both our sides
so that could be our smallest common element ?
wdyt ?
Lynn Foster
bernini's kanban is primarily a view of a VF structure, so it is not a standard kanban user-defined columns kind of thing
we are also in the process of adding a more traditional kanban view, kind of todo-doing-done thing with some ability to define columns within that - but it still will rely on the data inside the VF model
Tib Kat
ok but if we share actions , that's were your process can kick in
Lynn Foster
what's in an action? :smile:
(properties)
Tib Kat
we are into integrating , so we were about to integrate wekan
like rocket chat
Lynn Foster
did rocket chat connect to action? not following the thread :blush:
Tib Kat
{
    "_id" : ObjectId("57a8c1c1f6ca4712138b4595"),
    "email" : "oceatoon@gmail.com",
    "name" : "dzdzdzdzdz",
    "organizerId" : "57394d37f6ca4709228b4572",
    "organizerType" : "citoyens",
    "room" : "5786585cf6ca477b4e8b457d",
    "parentType" : "organizations",
    "parentId" : "54edb794f6b95c3c2a000941",
    "message" : "<p>dzdzdz<br></p>",
    "type" : "action",
    "startDate" : 1473451200.0000000000000000,
    "updated" : 1.47112e+09,
    "created" : 1.47068e+09,
    "viewCount" : 1,
    "links" : {
        "contributors" : {
            "57394d37f6ca4709228b4572" : {
                "type" : "citoyens",
                "isAdmin" : true
            }
        }
    },
    "comment" : {
        "57394d37f6ca4709228b4572" : [ 
            {
                "date" : ISODate("2016-08-13T19:05:51.000Z")
            }
        ]
    },
    "commentCount" : 1
}
that was about integrations
before today , we were thinking of fully integrating wekan as a organisations kanban
as we allready use it internally
it would have been a tool
but if we integrate a fully connected , and fully integrated kanban
wow that's even better
Lynn Foster
on your action, is it a plan, or are you recording something you did?
Tib Kat
it's open to interpretation
Lynn Foster
and is this basically a chat? or is this DDA?
Tib Kat
that's why that's were you guys can do what you do best
Lynn Foster
(both would be useful)
Tib Kat
take it from there
no the chat is just another tool
usable by the organization
DDA is Discuss Decide Act
Lynn Foster
can you show us a screen shot of that data example above?
Tib Kat
very loomio like approach at first
Lynn Foster
nice
Tib Kat
but you could propose a lot more productivity approch
let me get one
Lynn Foster
so i don't get more confused: are we talking about kanban integration or are we talking about DDA integration?
:smile:
Tib Kat
[Photo]
kanban into the DDA
but kanban could actually be a complete visualisation tool of everything in CO
like the graph view
shows things differently
Lynn Foster
could we take the code on the bottom part of that screen shot starting with "Discussion" and integrate it into bernini's kanban do you think?  is it a separate module now?
or does that data structure above cover this whole screen shot?
or maybe the integration point is the action could be the same as our commitment; or possibly our process
maybe room could be plan?  (plan is one kanban view for us, contains possibly many processes, which could be connected towards one deliverable, or many deliverables)
Tib Kat
hmmm that depends on a lot of things
Lynn Foster
:smile:
Tib Kat
Lynn Foster
or does that data structure above cover this whole screen shot?
yes
I think we're starting to go into detail here
Lynn Foster
indeed
Tib Kat
we'd have to show each other screens
and sharing points
Lynn Foster
where the devil is, as always :joy:
should we do this another time? in a call or something?
Tib Kat
not sure writting it all here is the best brainstorming process
Lynn Foster
probably not, it gets lost
Tib Kat
it's still early here
8:45PM
Lynn Foster
us too
oh time of day :smile:
8:45pm is early??? :smile:
Tib Kat
:)
Lynn Foster
like you still got kids up?
Tib Kat
nope that's dealt with
@Bernin1 are you in for a call ?
finding :
linking points
project connectors
I think we're on to something here :)
Lynn Foster
if bernini doesn't come back soon, let's wait and schedule something
Tib Kat
we can start thinking ? maybe ? if you guys are in sync ?
I'd love to start my week this way , the team would love this kind of news
Lynn Foster
we are in sync, but i'm not in touch at all with the technical side of his React UI code
Tib Kat
specifically integrated kanban + Value Flows implementations extending actions
so we could talk about features then , but indeed would be good to have him here , so nothing is lost and everything is gained
Lynn Foster
yeah, much better to have him here too
i'll put up some screen shots of his stuff here though as food for thought
Tib Kat
let's wait a bit then
Lynn Foster
yeah
hmmm, can't figure out how to run the kanban any more
Tib Kat
isn't it running online ?
as a demo ?
agent.fair.coop ?
Lynn Foster
it was running online - but now i am only getting the dashboard demo, can't seem to get to a kanban
ha found it
[Photo]
[Photo]
those are 2 different views of the same VF data - resource flow view and status view
we can get these from the existing data; want to add tags/additional columns to the last one
Tib Kat
so when you commit to something it goes to your todo ?
we have a "gather it all menu" like this
Lynn Foster
actually when you commit it goes into your doing as of now
we could have 2 columns, committed and doing, but currently we don't
todo has nothing committed and nothing logged for work
Tib Kat
[Photo]
this how a user quick finds his activity
Lynn Foster
do actions relate to the propositions / resolutions?
Tib Kat
things to focus on
yes off course
Lynn Foster
:smile:
Tib Kat
but the link is very open for the moment
we 3 states of proposal 
Amendment phase
Voting phase 
Resolution phase
then according to the proposal it can call for action
that's where actions quick in
Lynn Foster
got it - so we approach from the opposite direction: we have things that we want to do (like actions), then there is discussion about each one - some of the people use gitlab, so it is just like github issues, but assumed to be related to some deliverable already agreed to be done
or at least already recorded as a todo
Tib Kat
:) exactly that's why linking would be great
Lynn Foster
and a specific question: can the resolution happen without voting?  faircoop does a consensus thing
(also, I'll capture all of this and put it in an issue somewhere so we don't lose it)
Bob Haugen
Very promising discussion, TIb + Lynn!
Tib Kat
consensus is evey one accepts == vote For something 
on proposals you just put the resolution only if 100% is obtained
it default at 50%
Lynn Foster
ah ok
Tib Kat
Lynn Foster
got it - so we approach from the opposite direction: we have things that we want to do (like actions), then there is discussion
it's identical with us
Lynn Foster
so to summarize: we could hook it together both ways:
1. start with your kanban of work, and pick a process or commitment to discuss, which brings you one discussion thread on one process or commitment (== action)
2. start with the view you showed above, and if an action is created from the discussion, it can go on the kanban
Tib Kat
after resolution, actions are listed , discussed using comments
Lynn Foster
oh ok
Tib Kat
and then someone can , take the task
Lynn Foster
yup
Tib Kat
it's the in his own list
Lynn Foster
:+1:
does one action have one person who can take the task, or multiple?
Tib Kat
multi

**Comment by oceatoon** [comment]
```
{
    "_id" : ObjectId("597877c9539f222f746b3d63"),
    "email" : "oceatoon@gmail.com",
    "name" : "proj act",
    "type" : "actions",
    "parentType" : "projects",
    "parentId" : "59391e17539f228c517f592c",
    "tags" : [ 
        ""
    ],
    "created" : NumberLong(1502306926),
    "updated" : NumberLong(1502306926),
    "modified" : ISODate("2017-08-09T19:28:46.000Z")
}
```
